components/omnibox/browser/brave_search_provider_unittest.cc

```diff
@@ -182,3 +248,18 @@ TEST_F(BraveSearchProviderTest,
   ASSERT_NO_FATAL_FAILURE(QueryForInputAndSetWYTMatch(u"hel", &wyt_match));
   ASSERT_EQ(1u, provider_->matches().size());
 }
+
+#if !BUILDFLAG(IS_IOS)
+// Check search suggestion is blocked when input and clipboard text is same.
+TEST_F(BraveSearchProviderTest, DontSendClipboardTextToSuggest) {
+  // Check it's not blocked.
+  ASSERT_NO_FATAL_FAILURE(QueryForInput(u"brave"));
+  EXPECT_TRUE(test_url_loader_factory_.IsPending("https://defaultturl2/brave"));
+
+  // Check it's blocked.
```

Comment:
I'd suggest another test case just to cover all cases:
```suggestion
  // Not blocked when clipboard cont is different from URL bar.
  test_clipboard_->WriteText("best browser");
  ASSERT_NO_FATAL_FAILURE(QueryForInput(u"brave"));
  EXPECT_TRUE(test_url_loader_factory_.IsPending("https://defaultturl2/brave"));

  // Blocked when clipboard content is the same.
```

Reply:
Perfect. Updated 👍🏼